app/controllers/templates_controller.rb

```diff
@@ -0,0 +1,13 @@
+class TemplatesController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
Put this in ApplicationController then use skip_before_action in the controller where it's not needed. (Deny then allow is a good rule of thumb to follow.)